backend/msgpack/conn_test.go

```diff
@@ -22,78 +22,149 @@ package msgpack
 
 import (
 	"errors"
+	"fmt"
+	"math"
 	"net"
 	"sync"
 	"testing"
 	"time"
 
 	"github.com/m3db/m3x/clock"
 
+	"github.com/leanovate/gopter"
+	"github.com/leanovate/gopter/gen"
+	"github.com/leanovate/gopter/prop"
 	"github.com/stretchr/testify/require"
 )
 
 const (
-	testFakeServerAddr  = "nonexistent"
-	testLocalServerAddr = "127.0.0.1:0"
+	testFakeServerAddr        = "nonexistent"
+	testLocalServerAddr       = "127.0.0.1:0"
+	testRandomSeeed           = 831992
+	testMinSuccessfulTests    = 1000
+	testReconnectThreshold    = 1024
+	testMaxReconnectThreshold = 8096
 )
 
 var (
 	errTestConnect = errors.New("connect error")
 	errTestWrite   = errors.New("write error")
 )
 
-func TestConnectionWriteNoReconnect(t *testing.T) {
-	conn := newConnection(testFakeServerAddr, testConnectionOptions())
-	conn.connectWithLockFn = func() error { return errTestConnect }
+func TestConnectionProperties(t *testing.T) {
+	params := gopter.DefaultTestParameters()
+	params.Rng.Seed(testRandomSeeed)
+	params.MinSuccessfulTests = testMinSuccessfulTests
+	props := gopter.NewProperties(params)
 
-	require.Equal(t, errNoActiveConnection, conn.Write(nil))
-	require.Equal(t, 1, conn.numFailures)
-}
+	props.Property(
+		`When the number of failures is less than or equal to the threshold and the time since last `+
```

Comment:
nit: don't need the string concat, raw quoted strings can be multiline

Reply:
Yea, so I actually did this on purpose so that I could break this line up, without having to include a newline in it. Admittedly a little too compulsive, long lines really bother me for some reason lol

Reply:
Did the newline make the test output look funky?

Reply:
Yea, there was a random newline in the middle of it because I think `gopter` does its own pretty printing. It looked like:

```
foooooooooooooooooooo
oooo
oooooooooo
```

instead of

```
fooooooooooooooooooooo
oooooooooooooooo
```

Reply:
Weird that it causes issues for this property, but the one on line 124 doesn't.

Reply:
I think it's just based on the line length. Admittedly, I was way past the point of wanting to fiddle with this when I did this.